### DISCUSSION TITLE
Expression and curry_flip

### DISCUSSION THREAD

**Discussion by matiboy** [ReactiveX/RxPY#681] [Ideas]
Following up on @dbrattli 's comment in PR #678 :

> another change we want to do is to borrow the curry_flip function from the Expression library to simplify all the operator implementations by removing all the nested functions. Take a look at https://github.com/cognitedata/Expression/blob/main/expression/collections/block.py#L602 Thus you can put source back as the first parameter, and curry-flip will move it as the last parameter curried (returning a function that takes it as the parameter). 
1. Expression requires Python 3.9 so we would need to indeed borrow (aka copy) the code.
2. ~~Trying to understand the goal of the change; readability? better typing?~~ re-reading the comment, not sure if this is related to `concat_map` or more generic?
3. I've written the code below; tests pass, but typing is not happy, I think because `merge` is typed with `Any`

```python

@curry_flip(1)
def concat_map(
    source: Observable[_T1], project: Mapper[_T1, Observable[_T2]]
) -> Callable[[Observable[_T1]], Observable[_T2]]:
  return source.pipe(map(project), merge(max_concurrent=1))

```

**Comment by dbrattli** [comment]
The problem above is the typing of `merge`. This PR shows how it can be done: #682. I.e it can be used for all operartors, not just `concat_map`. Not sure if we should use it in the `__init__.py` file or just in the operator specific files such as e.g `_map.py`, or both. We need to try it out. But it will remove a lot of clutter in the operator specific files.
